Modules/Quest/QuestEventHandler.lua

```diff
@@ -0,0 +1,352 @@
+    QuestieJourney:CompleteQuest(questId)
+end
+
+--- Fires when a quest is removed from the quest log. This includes turning it in and abandoning it.
+---@param questId number
+function _QuestEventHandler:QuestRemoved(questId)
+    print("[Quest Event] QUEST_REMOVED", questId)
+
+    if (not questLog[questId]) then
+        print("--> questLog[questId] was nil")
+        questLog[questId] = {}
+    end
+
+    -- QUEST_TURNED_IN was called before QUEST_REMOVED --> quest was turned in
+    if questLog[questId].state == QUEST_LOG_STATES.QUEST_TURNED_IN then
+        print("--> Quest", questId, "was turned in before. Nothing do to.")
+        return
+    end
+
+    -- QUEST_REMOVED can fire before QUEST_TURNED_IN. If QUEST_TURNED_IN is not called after X seconds the quest
+    -- was abandoned
+    questLog[questId] = {
+        state = QUEST_LOG_STATES.QUEST_REMOVED,
+        timer = C_Timer.NewTicker(1, function()
+            _QuestEventHandler:MarkQuestAsAbandoned(questId)
+            questLog[questId].timer = nil
+        end, 1)
+    }
+    skipNextUQLCEvent = true
+end
+
+---@param questId number
+function _QuestEventHandler:MarkQuestAsAbandoned(questId)
+    if questLog[questId].state == QUEST_LOG_STATES.QUEST_REMOVED then
+        print("--> Quest", questId, "was abandoned")
+        questLog[questId].state = QUEST_LOG_STATES.QUEST_ABANDONED
+
+        QuestieHash:RemoveQuestHash(questId)
+        QuestieQuest:AbandonedQuest(questId)
+        QuestieJourney:AbandonQuest(questId)
+    end
+end
+
+---Fires when the quest log changed in any way. This event fires very often!
+function _QuestEventHandler:QuestLogUpdate()
+    print("[Quest Event] QUEST_LOG_UPDATE")
+
+    local continueQueuing = true
+    -- Some of the other quest event didn't have the required information and ordered to wait for the next QLU.
+    -- We are now calling the function which the event added.
+    while continueQueuing and next(questLogUpdateQueue) do
+        continueQueuing = _QuestLogUpdateQueue:GetFirst()()
+    end
+
+    if doFullQuestLogScan then
+        _QuestEventHandler:UpdateAllQuests()
+        doFullQuestLogScan = false
+    end
+end
+
+--- Fires whenever a quest objective progressed
+---@param questId number
+function _QuestEventHandler:QuestWatchUpdate(questId)
+    print("[Quest Event] QUEST_WATCH_UPDATE", questId)
+
+    -- We do a full scan even though we have the questId because many QUEST_WATCH_UPDATE can fire before
+    -- a QUEST_LOG_UPDATE. Also not every QUEST_WATCH_UPDATE gets a single QUEST_LOG_UPDATE and doing a full
+    -- scan is less error prone
+    doFullQuestLogScan = true
+    skipNextUQLCEvent = true
+end
+
+--- Fires when an objective changed in the quest log of the unitTarget. The required data is not available yet though
+---@param unitTarget string
+function _QuestEventHandler:UnitQuestLogChanged(unitTarget)
+    print("[Quest Event] UNIT_QUEST_LOG_CHANGED", unitTarget)
+
+    if unitTarget ~= "player" then
```

Comment:
Maybe filter with unitTarget == "player" before calling the function at all. It is not interest for us if other group members quest log changes.

fyi: There is also now way to register event fire only for certain unit, but I am not sure can it be used for this and the functionality had some "small prints" on usage. So better not use it now.